### PR TITLE
Improve documentation of generated suspending methods

### LINKED ISSUE #121: Incorrect documentation for xxxAwait methods derived from fluent methods
The issue was originally filed in the Vert.x Cassandra Client repository https://github.com/vert-x3/vertx-cassandra-client/issues/23

Consider the following fluent and async method:

```java
  /**
   * Connect to a Cassandra service.
   *
   * @param connectHandler handler called when asynchronous connect call ends
   * @return current Cassandra client instance
   */
  @Fluent
  CassandraClient connect(Handler<AsyncResult<Void>> connectHandler);
```

When the corresponding `connectAwait` method is generated, the return type is no longer `CassandraClient`, but the kdoc is misleading:

```
suspend fun CassandraClient.connectAwait(): Unit
Connect to a Cassandra service.

Parameters
connectHandler - handler called when asynchronous connect call ends

Returncurrent
Cassandra client instance *
```

### PR DESCRIPTION
**Pull request by gmariotti**
I took a bit of time to refactor the code generation part, hopefully, it should now be more explicit what each block is doing.

A couple of things are still missing:
- Cleanup imports from unused directives (like `io.vertx.core.Handler`)
- Improve method documentation

What do you think @vietj and @okou19900722?

Closes #121

### CODE REVIEW

**Comment by vietj** [comment]
I think it would be good to split the PR in two commits:

1/ the changes done to the generator
2/ the changes to the generated code once you run the generator

that would ease reviews :-)

**Comment by gmariotti** [comment]
Makes sense. Then, I will open a new one just for the generator changes and reuse this one for the generated code

**Comment by vietj** [comment]
I think you can keep a single PR, then we can drill down in the individual commits

File: vertx-lang-kotlin-gen/src/main/java/io/vertx/lang/kotlin/KotlinCoroutineGenerator.java [hunk 1]
```diff
@@ -69,46 +71,169 @@ public String render(ClassModel model, int index, int size, Map<String, Object>
...
+    String objectName;
+    if (isExtensionMethod) {
+      objectName = "this";
+    } else {
+      objectName = className;
+    }
+    generateFunctionBody(method, objectName, returnType, writer);
+  }
+
+  private TypeInfo getReturnType(MethodInfo method, ParamInfo lastParam) {
```
Comment (vietj) [inline vertx-lang-kotlin-gen/src/main/java/io/vertx/lang/kotlin/KotlinCoroutineGenerator.java]: rename it `getAsyncTypeArgument` as this returns the type argument of an asynchronous method wether it's an handler or an asynchronous handler

Comment (gmariotti) [reply vertx-lang-kotlin-gen/src/main/java/io/vertx/lang/kotlin/KotlinCoroutineGenerator.java]: fixed

**Comment by gmariotti** [comment]
Thanks for the review @vietj. If you're fine with it, I will merge it and backport it to 3.7

**Pull request merged**